prboom2/src/d_deh.c

```diff
@@ -2712,7 +2719,7 @@ static void deh_procMisc(DEHFILE *fpin, FILE* fpout, char *line) // done
                                   idkfa_armor_class = (int)value;
                                 else
                                   if (!deh_strcasecmp(key,deh_misc[14]))  // BFG Cells/Shot
-                                    bfgcells = (int)value;
+                                    weaponinfo[MT_BFG].ammopershot = bfgcells = (int)value;
```

Comment:
I don't think we should set this here. I'd rather have the meaning of this dehacked field be the same as it has been, which is overwriting the bfgcells value. The way it is now, it doesn't set WIF_ENABLEAPS anyway, which could be confusing, and the alternative of setting it would have ramifications for compatibility. Better to let it rest as it has been.

Reply:
This is mainly here for forwards-compatibility with Eternity -- EE does this exact thing, so if we don't do the same here, there may be divergent behavior.

Reply:
I think that's something to adjust in ee then personally 🤷 
Having 2 fields that write the same value, but differently, seems like a dangerous path to take. And making an old key have a partially new meaning is also not safe to me.

Reply:
Let's do this:
1) Don't set ammopershot here, but set `bfgcells_modified = true`.
2) After processing the deh, do this:
```c
if (bfgcells_modified && weaponinfo[MT_BFG].intflags & WIF_ENABLEAPS && bfgcells != weaponinfo[MT_BFG].ammopershot)
  I_Error("Mismatch between bfgcells and bfg ammo per shot modifications! Check your dehacked.");
```

Reply:
I guess that doesn't solve the problem of using a new codepointer that will reference ammopershot though, is that the real problem? :thinking:

Reply:
If that's the case, then DO set ammopershot here, but I want to raise I_Error if someone tries to set both ammopershot in the weaponinfo and bfgcells and they don't match. As long as that happens I'm all aboard.

Reply:
https://github.com/kraflab/dsda-doom/pull/24/commits/70f7a405feaa440dd1fb8b504811a8723587ad8a took care of this here - we can drop it later if you think we should, but let's not get stuck on this point.